client/src/components/ProductCard/ProductCard.js

```diff
@@ -1,9 +1,20 @@
-import { Card } from "react-bootstrap";
-// import { BsHeart, BsHeartFill } from 'react-icons/bs';
+import { Card, OverlayTrigger, Tooltip } from "react-bootstrap";
 import { Link } from "react-router-dom";
 import Moment from "react-moment";
+import React, { useRef } from "react";
+
+function Date({ children }) {
```

Comment:
When adding the OverlayTrigger, I abstracted the date to a component for consistency and to prevent repeating code. This could be moved to a global site component to allow reuse of the date format across the repo.